src/Synapse/Mapper/InserterTrait.php

```diff
@@ -58,8 +58,11 @@ protected function insertRow(AbstractEntity $entity, array $values)
 
         $result = $statement->execute();
 
-        if (! $entity->getId()) {
-            $entity->setId($result->getGeneratedValue());
+        $entityValues = $entity->getArrayCopy();
+        if ($this->autoIncrementColumn && ! $entityValues[$this->autoIncrementColumn]) {
```

Comment:
I would submit that the second condition `&& ! $entityValues[$this->autoIncrementColumn]` should be omitted.